### ISSUE TITLE
Warning message when loading config file in Python 2

### ISSUE THREAD

**Issue by bellert** [dwavesystems/dwave-cloud-client#76]
The following message appears when loading a config file in Python 2:
```
/env/local/lib/python2.7/site-packages/dwave/cloud/config.py:114: DeprecationWarning: You passed a bytestring as `filenames`. This will not work on Python 3. Use `cp.read_file()` or switch to using Unicode strings across the board.
  files_read = config.read(filename)
```
Despite what the message says, the load succeeds in both Python 2 and 3.

**Comment by bellert** [comment]
```
(env) bellert@nb-000-1285:~/git_root/circuit-fault-diagnosis-demo$ python -V
Python 2.7.12
(env) bellert@nb-000-1285:~/git_root/circuit-fault-diagnosis-demo$ dwave ping
Using endpoint: https://cloud.dwavesys.com/sapi
Traceback (most recent call last):
  File "/home/bellert/git_root/circuit-fault-diagnosis-demo/env/bin/dwave", line 11, in <module>
    sys.exit(cli())
  File "/home/bellert/git_root/circuit-fault-diagnosis-demo/env/local/lib/python2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/bellert/git_root/circuit-fault-diagnosis-demo/env/local/lib/python2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/bellert/git_root/circuit-fault-diagnosis-demo/env/local/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/bellert/git_root/circuit-fault-diagnosis-demo/env/local/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/bellert/git_root/circuit-fault-diagnosis-demo/env/local/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/bellert/git_root/circuit-fault-diagnosis-demo/env/local/lib/python2.7/site-packages/dwave/cloud/cli.py", line 121, in ping
    print("Using solver: {}".format(solver.id))
UnboundLocalError: local variable 'solver' referenced before assignment
```

Contents of `dwave.conf`:
```
[defaults]
endpoint = https://cloud.dwavesys.com/sapi
client = qpu
token = ...

[DW_2000Q_VFYC_1]
solver = DW_2000Q_VFYC_1

[DW_2000Q_1]
solver = DW_2000Q_1

[BAY14_X_INTERNAL]
solver = BAY14_X_INTERNAL

[c4-sw_optimize]
client = cpu
solver = c4-sw_optimize

[c4-sw_sample]
solver = c4-sw_sample
```

**Comment by randomir** [comment]
Ok, this was a separate issue #78. I've just fixed that in #79 (minor fix).

Now `dwave ping` works, but I don't see the warning.. Can you maybe provide the sample code that reproduces this behaviour?

**Comment by bellert** [comment]
https://github.com/dwavesystems/circuit-fault-diagnosis-demo/blob/master/demo.py

Relevant lines:
```python
from dwave.system.samplers import DWaveSampler
from dwave.system.composites import EmbeddingComposite
NUM_READS = 1000
bqm, labels = three_bit_multiplier() # constructs the problem to be sent to the QPU
sampler = EmbeddingComposite(DWaveSampler())
response = sampler.sample_ising(bqm.linear, bqm.quadratic, num_reads=NUM_READS)
```
